0000-catalog.md

```diff
@@ -0,0 +1,789 @@
+should be performed in a payload.
+
+### CatalogPayload Transaction
+
+```
+message CatalogPayload {
+    enum Actions {
+        UNSET_ACTION = 0;
+        CATALOG_CREATE = 1;
+        CATALOG_UPDATE = 2;
+        CATALOG_DELETE = 3;
+        CATALOG_PRODUCT_CREATE = 100;
+        CATALOG_PRODUCT_UPDATE = 101;
+        CATALOG_PRODUCT_DELETE = 102;
+        CATALOG_PRODUCT_SET_STATUS = 103;
+    }
+
+    Action action = 1;
+
+    // Approximately when transaction was submitted, as a Unix UTC timestamp
+    uint64 timestamp = 2;
+
+    CatalogCreateAction catalog_create = 3;
+    CatalogUpdateAction catalog_update = 4;
+    CatalogDeleteAction catalog_delete = 5;
+    CatalogProductCreateAction catalog_product_create = 100;
+    CatalogProductUpdateAction catalog_product_update = 101;
+    CatalogProductDeleteAction catalog_product_delete = 102;
+    CatalogProductSetStatusAction set_catalog_product_status = 103;
+}
+```
+
+## Catalog Actions
+
+### CatalogCreateAction
+
+CatalogCreateAction adds a new catalog to state. The transaction should be
+submitted by an agent, which is identified by its signing key, acting on behalf 
+of the organization that corresponds to the owner in the create transaction. 
+(Organizations and agents are defined by the Pike smart contract.)
+
+```
+message CatalogCreateAction {
+    // GS1 Company Prefix from owner + catalog_id are use as
+    // a composite key for determining the state address
+    string owner = 1;
+    string catalog_id = 2;
+    string name = 3;
+    repeated PropertyValues properties = 4;
+}
+```
+
+Validation requirements:
+
+- If a catalog with catalog_id already exists the transaction is invalid.
+- The signer of the transaction must be an agent in the Pike state and must
+  belong to an organization in Pike state, otherwise the transaction is invalid.
+- The agent must have the permission can_create_catalog for the organization,
+  otherwise the transaction is invalid.
+
+The inputs for CatalogCreateAction must include:
+
+- Address of the Agent submitting the transaction
+- Address of the Organization the Catalog is being created for
+- Address of the Catalog to be created
+
+The outputs for CatalogCreateAction must include:
+
+Address of the Catalog created
+
+### CatalogUpdateAction
+
+CatalogUpdateAction updates an existing catalog in state. The transaction should be
+submitted by an agent, which is identified by its signing key, acting on behalf 
+of the organization that corresponds to the owner in the update transaction. 
+(Organizations and agents are defined by the Pike smart contract.)
+
+```
+message CatalogUpdateAction {
+    // GS1 Company Prefix from owner + catalog_id are use as
+    // a composite key for determining the state address
+    string owner = 1;
+    string catalog_id = 2;
+    string name = 3;
+    repeated PropertyValues properties = 4;
+}
+```
+
+Validation requirements:
+
+- If a catalog with catalog_id exists the transaction is invalid.
```

Comment:
I don't think this is what you want ... if the `catalog_id` does not exist, how would the entity be updated?

Reply:
You are correct... typo, thanks! ✅